packages/react/src/components/DataTable/DataTable.js

```diff
@@ -310,8 +310,8 @@ export default class DataTable extends React.Component {
 
     // Otherwise, we're working on `TableSelectAll` which handles toggling the
     // selection state of all rows.
-    const rowCount = this.state.rowIds.length;
-    const selectedRowCount = this.getSelectedRows().length;
+    const rowCount = this.getFilteredRowIds().length;
+    const selectedRowCount = this.getFilteredAndSelectedRowIds().length;
```

Comment:
These seem to be redundant calls in the use case where a data table is using selection without filtering. Perhaps a ternary can be used here?